arduino/cores/packagemanager/loader.go

```diff
@@ -356,9 +356,62 @@ func (pm *PackageManager) loadPlatformRelease(platform *cores.PlatformRelease, p
 		return fmt.Errorf(tr("loading boards: %s"), err)
 	}
 
+	if !platform.PluggableDiscoveryAware {
+		convertLegacyPlatformToPluggableDiscovery(platform)
+	}
```

Comment:
I'm strongly against this. 

This is not being retrocompatible, we never were compatible with those platforms, this is adding a new feature to the CLI that will remain here forever and will need to be maintained.

Platform maintainers will keep using the old syntax if we do this, that syntax is not even documented. We have a new specification to support network upload and we must push platform maintaners to use it, if we keep supporting old things it will be harder and harder to move forward with newer ones.

Reply:
IMHO we are not creating legacy but supporting the existing ecosystem. If we remove this function the network upload, already supported by the Java IDE, will not work with the Arduino CLI on all the currently existing platforms (at least until the 3rd party developers adopt the pluggable discovery and make a new upstream release).

IMHO we should start by supporting the current platforms and leave the possibility to the upstream maintainers to upgrade at their own pace. They will eventually upgrade to pluggable discovery because it's a clearly superior specification, but until then the users will benefit from the network upload with all the currently available platforms.

Reply:
I don't think it's a good idea but since @cmaglie and I couldn't settle this debate @rsora did it for us and we're going forward with the conversion.

Nonetheless I'd like some unit tests for the `convertLegacyPlatformToPluggableDiscovery` function.